src/app/manager-dashboard/reports/reports-detail.component.ts

```diff
@@ -270,7 +270,13 @@ export class ReportsDetailComponent implements OnInit, OnDestroy {
   }
 
   openExportDialog(reportType: 'logins' | 'resourceViews' | 'courseViews' | 'summary') {
-    const minDate = new Date(this.activityService.minTime(this.loginActivities.data, 'loginTime')).setHours(0, 0, 0, 0);
+    let minDate = new Date(this.activityService.minTime(this.loginActivities.data, 'loginTime')).setHours(0, 0, 0, 0);
+
+    const startDateFromForm: Date = new Date(this.dateFilterForm.controls.startDate.value);
+    if (startDateFromForm !== this.minDate) {

```

Comment:
This statement will always be `true`.  An Object in JavaScript is always a reference, and when you create an object (with `new` or `const object = { ... }`) it creates a new reference.  When you compare an object with `===`, `!==`, `==`, or `!=` only the references are compared, not the values.

Regardless, you do not need to check if the form value is the same as minDate.  Either way you should set the values to the form value.

Reply:
Thank you, for your review. I have made the changes.